src/main/java/com/artipie/asto/etcd/EtcdStorage.java

```diff
@@ -67,13 +71,28 @@ public EtcdStorage(final Client client) {
 
     @Override
     public CompletableFuture<Collection<Key>> list(final Key prefix) {
-        return this.client.getKVClient().get(
-            keyToSeq(prefix),
-            GetOption.newBuilder()
-                .withKeysOnly(true)
-                .withSortOrder(SortOrder.ASCEND)
-                .build()
-        ).thenApply(
+        final CompletableFuture<GetResponse> future;
+        if (prefix == Key.ROOT) {
+            final ByteSequence root = keyToSeq(new Key.From("\0"));
```

Comment:
There's no need to wrap a key here - we can just create private constant of byteSequence